File: webapp/src/model/address.js
export default class {
    constructor(codAddress, city, neighborhood, street, number) {
        this.codAddress = codAddress;
        this.city = city;
        this.neighborhood = neighborhood;
        this.street = street;
        this.number = number;
    }
    toString(){
        return `${this.street}, ${this.number}, ${this.neighborhood}  - ${this.city}`;
    }
}

File: src/main/java/br/ufsc/cultivar/model/Address.java
package br.ufsc.cultivar.model;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Value;
import lombok.experimental.FieldDefaults;
import lombok.experimental.Wither;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Value
@Wither
@Builder(builderClassName = "Builder")
@JsonDeserialize(builder = Address.Builder.class)
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
public class Address {
    Long codAddress;
    @NotBlank
    @NotNull
    String city;
    @NotBlank
    @NotNull
    String neighborhood;
    @NotBlank
    @NotNull
    String street;
    String number;

    @JsonPOJOBuilder(withPrefix = "")
    public static class Builder{
        public Builder(){}
    }
}
